jni/include/faiss_stream_support.h

```diff
@@ -56,6 +56,40 @@ class FaissOpenSearchIOReader final : public faiss::IOReader {
 };  // class FaissOpenSearchIOReader
 
 
+/**
+ * A glue component inheriting IOWriter to delegate IO processing down to the given
+ * mediator. The mediator is expected to do write bytes via the provided Lucene's IndexOutput.
+ */
+class FaissOpenSearchIOWriter final : public faiss::IOWriter {
+ public:
+  explicit FaissOpenSearchIOWriter(NativeEngineIndexOutputMediator *_mediator)
+      : faiss::IOWriter(),
+        mediator(_mediator) {
+    name = "FaissOpenSearchIOWriter";
+  }
+
+  size_t operator()(const void *ptr, size_t size, size_t nitems) final {
+    const auto writeBytes = size * nitems;
+    if (writeBytes > 0) {
+      mediator->writeBytes((uint8_t *) ptr, writeBytes);
```

Comment:
Any particular reason to not use static_cast instead of the C-style cast when converting the ptr to uint8_t *?
```
mediator->writeBytes(static_cast<const uint8_t *>(ptr), writeBytes);

```

Reply:
will update in next rev.